### ISSUE TITLE
check if DataFrames.jl is affected by Threads.@spawn bug

### ISSUE THREAD

**Issue by bkamins** [JuliaData/DataFrames.jl#3207]
x-ref https://github.com/JuliaLang/julia/issues/40626, https://github.com/JuliaData/CSV.jl/pull/1046, https://github.com/JuliaData/CSV.jl/issues/1045

**Comment by bkamins** [comment]
I have checked and DataFrames.jl is affected by this issue.

**Comment by bkamins** [comment]
The question is if running this function after every `@sync` of spawned tasks:
```
function _clear_thread_state()
    Threads.@threads :static for _ in 1:Threads.nthreads()
        nothing
    end
end
```
is good enough to ensure clearing thread state and will not cause problems in case some other tasks are spawned independently what DataFrames.jl does.

CC @tkf @nalimilan @quinnj

**Comment by quinnj** [comment]
PR to CSV.jl with an alternative approach that I think may be more all-around correct: https://github.com/JuliaData/CSV.jl/pull/1058 (it links to 2 other issues in JuliaLang/julia & CSV.jl where I discuss recent findings/thoughts in more detail).

**Comment by quinnj** [comment]
Yes, that would avoid holding references to the return value, as far as I understand.

**Comment by vchuravy** [comment]
Yes that would suffice, if you can guarantee that only one user will call `fetch` alternativly you could use a 1-element channel. Which would have the additional benefit of being potentially type-stable.
